tests/unit/s2n_client_key_share_extension_test.c

```diff
@@ -88,43 +88,48 @@ int main(int argc, char **argv)
         }
 
-        /* Test that s2n_client_key_share_extension.send writes a well-formed list of key shares */
+        /* Test that s2n_client_key_share_extension.send writes a keyshare for the highest priority 
+         * supported group in the ecc_preferences list */
         {
             struct s2n_stuffer key_share_extension;
             struct s2n_connection *conn;
             EXPECT_NOT_NULL(conn = s2n_connection_new(S2N_CLIENT));
             EXPECT_SUCCESS(s2n_stuffer_growable_alloc(&key_share_extension, 0));
             EXPECT_SUCCESS(s2n_client_key_share_extension.send(conn, &key_share_extension));
 
-            /* should have correct shares size */
-            uint16_t key_shares_size;
-            EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &key_shares_size));
-            uint16_t actual_key_shares_size = s2n_stuffer_data_available(&key_share_extension);
-            EXPECT_EQUAL(key_shares_size, actual_key_shares_size);
-            EXPECT_EQUAL(key_shares_size, s2n_stuffer_data_available(&key_share_extension));
-
             const struct s2n_ecc_preferences *ecc_preferences = NULL;
             EXPECT_SUCCESS(s2n_connection_get_ecc_preferences(conn, &ecc_preferences));
             EXPECT_NOT_NULL(ecc_preferences);
 
-            /* should contain every supported curve, in order, with their sizes */
-            for (int i = 0; i < ecc_preferences->count; i++) {
-                uint16_t iana_value, share_size;
-                EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &iana_value));
-                EXPECT_EQUAL(iana_value, ecc_preferences->ecc_curves[i]->iana_id);
-                EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &share_size));
-                EXPECT_EQUAL(share_size, ecc_preferences->ecc_curves[i]->share_size);
+            uint16_t key_shares_size;
+            EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &key_shares_size));
+            EXPECT_EQUAL(s2n_stuffer_data_available(&key_share_extension), key_shares_size);
 
-                EXPECT_SUCCESS(s2n_stuffer_skip_read(&key_share_extension, share_size));
-            }
+            uint32_t bytes_processed = 0;
+            EXPECT_EQUAL(key_shares_size, ecc_preferences->ecc_curves[0]->share_size + S2N_SIZE_OF_NAMED_GROUP + S2N_SIZE_OF_KEY_SHARE_SIZE);
+
+            uint16_t iana_value, share_size;
+            EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &iana_value));
+            EXPECT_SUCCESS(s2n_stuffer_read_uint16(&key_share_extension, &share_size));
+            bytes_processed += share_size + S2N_SIZE_OF_NAMED_GROUP + S2N_SIZE_OF_KEY_SHARE_SIZE;
+
+            EXPECT_EQUAL(iana_value, ecc_preferences->ecc_curves[0]->iana_id);
+            EXPECT_EQUAL(share_size, ecc_preferences->ecc_curves[0]->share_size);
+            EXPECT_SUCCESS(s2n_stuffer_skip_read(&key_share_extension, share_size));
+            EXPECT_EQUAL(bytes_processed, key_shares_size);
```

Comment:
Did you mean to verify share_size here? Since `key_shares_size` hasn't been modified since the last `EXPECT_EQUAL` I'm not sure what the intention is. Can the variable be grouped a little closer together?

Reply:
`key_shares_size` is not modified but `bytes_processed` is updated to include the `share_size` obtained since the last `EXPECT_EQUAL` on line 122. 

Here we intend to check that all the bytes are processed.